app-text/uudeview/files/uudeview-0.5.20-fix-build-for-clang16.patch

```diff
@@ -0,0 +1,221 @@
+         * fails.
+         */
+        if( (hp = gethostbyname( machine ) ) == NULL ) {
+-               unsigned long inet_addr();
+                static struct hostent def;
+                static struct in_addr defaddr;
+                static char *alist[1];
+@@ -344,8 +346,8 @@ char	*machine;
+  *	Errors:		Printed via nerror.
+  */
+ 
+-get_dnet_socket(machine)
+-char	*machine;
++int 
++get_dnet_socket (char *machine)
+ {
+ 	int	s, area, node;
+ 	struct	sockaddr_dn sdn;
+@@ -427,9 +429,8 @@ char	*machine;
+  *	Side effects:	None.
+  */
+ 
+-handle_server_response(response, server)
+-int	response;
+-char	*server;
++int 
++handle_server_response (int response, char *server)
+ {
+     switch (response) {
+ 	case OK_NOPOST:		/* fall through */
+@@ -477,9 +478,8 @@ char	*server;
+  *			fflush.
+  */
+ 
+-void
+-put_server(string)
+-char *string;
++void 
++put_server (char *string)
+ {
+ #ifdef DEBUG
+ 	fprintf(stderr, ">>> %s\n", string);
+@@ -502,9 +502,8 @@ char *string;
+  *	Side effects:	Talks to server, changes contents of "string".
+  */
+ 
+-get_server(string, size)
+-char	*string;
+-int	size;
++int
++get_server (char *string, int size)
+ {
+ 	register char *cp;
+ 
+@@ -536,8 +535,8 @@ int	size;
+  *			after this routine is called.
+  */
+ 
+-void
+-close_server()
++void 
++close_server (void)
+ {
+ 	char	ser_line[256];
+ 
+--- a/inews/clientlib.h
++++ b/inews/clientlib.h
+@@ -9,3 +9,7 @@ extern	int	server_init();
+ extern	void	put_server();
+ extern	int	get_server();
+ extern	void	close_server();
++
++extern int get_tcp_socket (char *machine);
++extern int get_server (char *string, int size);
++extern int handle_server_response (int response, char *server);
+--- a/inews/inews.c
++++ b/inews/inews.c
+@@ -1,3 +1,7 @@
++int strneql (register char *a, register char *b, int n);
++int gen_frompath (void);
++int valid_header (register char *h);
++
+ #ifndef lint
+ static char *sccsid = "@(#)inews.c	1.16	(Berkeley) 8/27/89";
+ #endif
+@@ -39,7 +43,7 @@ static char *sccsid = "@(#)inews.c	1.16	(Berkeley) 8/27/89";
+ 
+ #include "conf.h"
+ #include "nntp.h"
+-
++#include "clientlib.h"
+ 
+ #define	MAX_SIGNATURE	6
+ 
+@@ -47,9 +51,8 @@ extern	FILE	*ser_wr_fp;
+ 
+ char	host_name[256];
+ 
+-main(argc, argv)
+-int	argc;
+-char	*argv[];
++int 
++main (int argc, char *argv[])
+ {
+ 	char	line[NNTP_STRLEN], s[NNTP_STRLEN];
+ 	int	seen_fromline, in_header, seen_header;
+@@ -181,7 +184,8 @@ char	*argv[];
+  * The rn-style DOTDIR environmental variable is used if present.
+  */
+ 
+-void append_signature()
++void 
++append_signature (void)
+ {
+ 	char	line[256], sigfile[256];
+ 	char	*cp;
+@@ -254,7 +258,8 @@ void append_signature()
+  * a From: line in it.
+  */
+ 
+-gen_frompath()
++int 
++gen_frompath (void)
+ {
+ 	char	*full_name;
+ 	char	*cp;
+@@ -330,9 +335,8 @@ gen_frompath()
+  *	Side effects:	None.
+  */
+ 
+-strneql(a, b, n)
+-register char *a, *b;
+-int	n;
++int 
++strneql (register char *a, register char *b, int n)
```

Comment:
Please drop register whenever you see it, if you're already touching the line. It does nothing and hopefully it'll be removed from C one day.

Reply:
Let's see if we can automate that.

Reply:
hanno will do it later himselv